nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/test/java/org/apache/nifi/processors/aws/cloudwatch/TestPutCloudWatchMetric.java

```diff
@@ -264,27 +265,34 @@ public void testMetricExpressionInvalidRoutesToFailure() throws Exception {
         runner.enqueue(new byte[] {}, attributes);
         runner.run();
 
-        assertEquals(0, mockPutCloudWatchMetric.putMetricDataCallCount);
+        Assert.assertEquals(0, mockPutCloudWatchMetric.putMetricDataCallCount);
         runner.assertAllFlowFilesTransferred(PutCloudWatchMetric.REL_FAILURE, 1);
     }
 
-    @Test
-    public void testInvalidUnitRoutesToFailure() throws Exception {
+    @ParameterizedTest
+    @CsvSource({"nan","percent","count"})
+    public void testInvalidUnit(String unit) throws Exception {
         MockPutCloudWatchMetric mockPutCloudWatchMetric = new MockPutCloudWatchMetric();
-        mockPutCloudWatchMetric.throwException = new InvalidParameterValueException("Unit error message");
         final TestRunner runner = TestRunners.newTestRunner(mockPutCloudWatchMetric);
 
         runner.setProperty(PutCloudWatchMetric.NAMESPACE, "TestNamespace");
         runner.setProperty(PutCloudWatchMetric.METRIC_NAME, "TestMetric");
-        runner.setProperty(PutCloudWatchMetric.UNIT, "BogusUnit");
-        runner.setProperty(PutCloudWatchMetric.VALUE, "1");
-        runner.assertValid();
+        runner.setProperty(PutCloudWatchMetric.UNIT, unit);
+        runner.setProperty(PutCloudWatchMetric.VALUE, "1.0");
+        runner.assertNotValid();
+    }
 
-        runner.enqueue(new byte[] {});
-        runner.run();
+    @ParameterizedTest
+    @CsvSource({"Count","Bytes","Percent"})
```

Comment:
Minor, but here I'd reference the collection created in the processor. This way these values would be tied to the valid ones and we could have the benefit that this test could run for all valid units.

Reply:
Good point. I'll work on this!

Reply:
How would you like me to import the Unit set from the PutCloudWatch?  I just made a getter() for it.

Reply:
Probably I'd do something like that. This way every case will run as a separate test.
```
@ParameterizedTest
@MethodSource("data")
public void testValidUnit(String unit) {
    MockPutCloudWatchMetric mockPutCloudWatchMetric = new MockPutCloudWatchMetric();
    final TestRunner runner = TestRunners.newTestRunner(mockPutCloudWatchMetric);

    runner.setProperty(PutCloudWatchMetric.NAMESPACE, "TestNamespace");
    runner.setProperty(PutCloudWatchMetric.METRIC_NAME, "TestMetric");
    runner.setProperty(PutCloudWatchMetric.UNIT, unit);
    runner.setProperty(PutCloudWatchMetric.VALUE, "1");
    runner.assertValid();
}

private static Stream<Arguments> data() {
    return PutCloudWatchMetric.UNITS.stream().map(Arguments::of);
}
```
You can notice that I changed the units to public, but getter is also fine.

Reply:
Wicked, never seen that syntax with the @MethodSource, ill implement that.

Reply:
Done :)